Implement user role binding phase

mikenairn opened the pull request:
* Adds a role binding allowing the user defined in the walkthrough cr to view the namespace created.
* Generate namespace name based on the username, removed "namespace" from the crd.
* Add labels to resources created by this operator.

**Verification**

Run the operator locally:

```make run```

Create a user that has no elevated permissions:

```oc create user testuser --full-name=testuser```

Create a new walkthrough crd:

```.yml
apiVersion: "integreatly.aerogear.org/v1alpha1"
kind: "Walkthrough"
metadata:
  name: "testuserrole"
spec:
  username: testuser
  services:
   - fuse
```
```
kubectl create -f testuserrole.yaml -n integreatly
```

Verify that when you login as the  "testuser" user you can view the "testuser-walkthroughs" namespace.

mikenairn committed:
Implement user role binding phase

mikenairn commented on pkg/stub/handler.go:
```diff
@@ -80,12 +81,26 @@ func (h *Handler) provisionNamespace(wt *v1alpha1.Walkthrough) (*v1alpha1.Walkth
+func (h *Handler) userRoleBindings(wt *v1alpha1.Walkthrough) (*v1alpha1.Walkthrough, error) {
+	wtCopy := wt.DeepCopy()
+
+	userRoles := []string{"view", "edit"}
```
@maleck13 What roles do we want the user to have on this namespace?

maleck13 replied:
edit should be enough I think

maleck13 commented on pkg/stub/handler.go:
```diff
@@ -97,8 +112,27 @@ func (h *Handler) provisionServices(wt *v1alpha1.Walkthrough) (*v1alpha1.Walkthr
+	return &rbacv1.RoleBinding{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "RoleBinding",
+			APIVersion: "rbac.authorization.k8s.io/v1",
```
might be worth adding a label so we know it is our secret, that way we can easily find it again in the future if needed

mikenairn replied:
Labels are added to both namspaces and role bindings created by this operator here https://github.com/integr8ly/walkthrough-operator/pull/3/commits/dfab48c0961706050cebab0e874ad180e56c2df0

maleck13 reviewed (commented):
you may have mentioned this before. But I think the operator should decide the namespace name based on the walkthrough name ```GenerateName: walkthrough.Name + "-"```. There could be a security concern with giving roles to a user when they get to control the namespace name.
https://github.com/integr8ly/walkthrough-operator/pull/3/files#diff-2d7ceed365f1ea47bba386dac808cb07R80

mikenairn committed:
Generate namespace name based on username
Update cr status with namespace
Add labels to role bindings and namespaces created by operator.

mikenairn commented:
@maleck13 @aidenkeating Modified this to generate the namespace and removed it from the spec. see https://github.com/integr8ly/walkthrough-operator/pull/3/commits/dfab48c0961706050cebab0e874ad180e56c2df0

Generated namespace is set in the cr status : 
```
status:
  namespace: developer-walkthroughs
```

aidenkeating commented:
@mikenairn I'm going to give this a run
